src/System.Globalization/tests/TextElementEnumerator/TextElementEnumeratorTests.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class TextElementEnumeratorMoveNext
+    {
+        [Fact]
+        public void Enumerate()
+        {
+            string[] expectedElements = new string[] { "\uD800\uDC00", "\uD800\uDC00", "\u0061\u0300", "\u0061\u0300", "\u00C6" };
+            int[] expectedElementIndices = new int[] { 0, 2, 4 };
+            TextElementEnumerator enumerator = StringInfo.GetTextElementEnumerator("\uD800\uDC00\u0061\u0300\u00C6");
+
+            for (int i = 0; i < 2; i++)
```

Comment:
The previous `PosTest` test only tested after calling Reset; so I like that this version uses the enumerator freshly.

I like a little less that there is no test which does the work without calling Reset; which means that if this test fails it's hard to identify if it was on the first or second pass.